src/gso.ts

```diff
@@ -24,10 +22,47 @@ import {
 } from '@dual-finance/staking-options';
 import { getAssociatedTokenAddress } from '@project-serum/associated-token';
 import gsoIdl from './gso.json';
+import { parseGsoState } from './utils';
 
-export const GSO_PK: PublicKey = new PublicKey('DuALd6fooWzVDkaTsQzDAxPGYCnLrnWamdNNTNxicdX8');
+const GSO_STATE_SIZE = 1000;
+export const GSO_PK: PublicKey = new PublicKey(
+  'DuALd6fooWzVDkaTsQzDAxPGYCnLrnWamdNNTNxicdX8',
+);
 const metaplexId = new PublicKey('metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s');
 
+interface SOState {
+  soName: string;
+  authority: PublicKey;
+  optionsAvailable: number;
+  optionExpiration: number;
+  subscriptionPeriodEnd: number;
+  baseDecimals: number;
+  quoteDecimals: number;
+  baseMint: PublicKey;
+  quoteMint: PublicKey;
+  quoteAccount: PublicKey;
+  lotSize: number;
+  stateBump: number;
+  vaultBump: number;
+  strikes: number[];
+}
+export type GsoParams = {
+  periodNum: number;
+  subscriptionPeriodEnd: number;
+  lockupRatioTokensPerMillion: number;
+  strike: number;
+  projectName: string;
+  stakingOptionsState: PublicKey;
+  authority: PublicKey;
+  baseMint: PublicKey;
+  lockupPeriodEnd: number;
+  publicKey: PublicKey;
```

Comment:
publicKey for what? the gso state?

Reply:
Note that in the variable name. Should be obvious from reading exactly what it is